tls/s2n_resume.c

```diff
@@ -197,33 +195,30 @@ static int s2n_client_deserialize_resumption_state(struct s2n_connection *conn,
 
 int s2n_resume_from_cache(struct s2n_connection *conn)
 {
-    uint8_t data[S2N_STATE_SIZE_IN_BYTES] = { 0 };
-    struct s2n_blob entry = {.data = data,.size = S2N_STATE_SIZE_IN_BYTES };
+    uint8_t data[S2N_TICKET_SIZE_IN_BYTES] = { 0 };
+    struct s2n_blob entry = {.data = data,.size = S2N_TICKET_SIZE_IN_BYTES };
     struct s2n_stuffer from = {0};
     uint64_t size;
 
     S2N_ERROR_IF(conn->session_id_len == 0, S2N_ERR_SESSION_ID_TOO_SHORT);
     S2N_ERROR_IF(conn->session_id_len > S2N_TLS_SESSION_ID_MAX_LEN, S2N_ERR_SESSION_ID_TOO_LONG);
+    size = entry.size;
 
-    GUARD(s2n_stuffer_init(&from, &entry));
-    uint8_t *state = s2n_stuffer_raw_write(&from, entry.size);
-    notnull_check(state);
-
-    size = S2N_STATE_SIZE_IN_BYTES;
+    GUARD_NONBLOCKING(conn->config->cache_retrieve(conn, conn->config->cache_retrieve_data, conn->session_id, conn->session_id_len, entry.data, &size));
 
-    /* cache_retrieve relies on the caller to return S2N_CALLBACK_BLOCKED if data is not available */
-    GUARD_NONBLOCKING(conn->config->cache_retrieve(conn, conn->config->cache_retrieve_data, conn->session_id, conn->session_id_len, state, &size));
-
-    S2N_ERROR_IF(size != S2N_STATE_SIZE_IN_BYTES, S2N_ERR_SIZE_MISMATCH);
-    GUARD(s2n_deserialize_resumption_state(conn, &from));
+    
+    S2N_ERROR_IF(size != entry.size, S2N_ERR_SIZE_MISMATCH);
+    GUARD(s2n_stuffer_init(&from, &entry));
+    GUARD(s2n_stuffer_write(&from, &entry));
+    GUARD(s2n_decrypt_session_cache(conn, &from));
 
-    return S2N_SUCCESS;
+    return 0;
```

Comment:
why this change?

Reply:
Make it consistent with return code from other functions.